overlord/hookstate/ctlcmd/model.go

```diff
@@ -0,0 +1,187 @@
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/snapcore/snapd/client/clientutil"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+var (
+	shortModelHelp = i18n.G("Get the active model for this device")
+	longModelHelp  = i18n.G(`
+The model command returns the active model assertion information for this
+device.
+
+By default, the model identification information is presented in a structured,
+yaml-like format, but this can be changed to json by using the --json flag.
+
+Similarly, the active serial assertion can be used for the output instead of the
+model assertion.
+`)
+)
+
+func init() {
+	addCommand("model", shortModelHelp, longModelHelp, func() command { return &modelCommand{} })
+}
+
+type modelCommand struct {
+	baseCommand
+	Assertion bool `long:"assertion"`
+	Json      bool `long:"json"`
+	TermWidth int  `long:"term-width"`
+}
+
+type modelCommandFormatter struct {
+	storeAccount *snap.StoreAccount
+}
+
+// GetEscapedDash must be implemented as a part of the interface
+func (mf modelCommandFormatter) GetEscapedDash() string {
+	return "--"
+}
+
+// LongPublisher must be implemented as a part of the interface
+func (mf modelCommandFormatter) LongPublisher(storeAccountID string) string {
+	if mf.storeAccount == nil {
+		return mf.GetEscapedDash() + "\033[0m"
+	}
+
+	badge := ""
+	if mf.storeAccount.Validation == "verified" {
+		badge = "*"
+	}
+
+	// NOTE this makes e.g. 'Potato' == 'potato', and 'Potato Team' == 'potato-team',
+	// but 'Potato Team' != 'potatoteam', 'Potato Inc.' != 'potato' (in fact 'Potato Inc.' != 'potato-inc')
+	if strings.EqualFold(strings.Replace(mf.storeAccount.Username, "-", " ", -1), mf.storeAccount.DisplayName) {
+		return mf.storeAccount.DisplayName + badge + "\033[0m"
+	}
+	return fmt.Sprintf("%s (%s%s%s)", mf.storeAccount.DisplayName, mf.storeAccount.Username, badge, "\033[0m")
+}
+
+func newTabWriter(output io.Writer) *tabwriter.Writer {
+	minWidth := 2
+	tabWidth := 2
+	padding := 2
+	padchar := byte(' ')
+	return tabwriter.NewWriter(output, minWidth, tabWidth, padding, padchar, 0)
+}
+
+// reportError prints the error message to stderr
+func (c *modelCommand) reportError(format string, a ...interface{}) {
+	w := newTabWriter(c.stderr)
+	fmt.Fprintf(w, format, a...)
+	w.Flush()
+}
+
+// checkGadgetOrModel verifies that the current snap context is either a gadget snap or that
+// the snap shares the same publisher as the current model assertion.
+func (c *modelCommand) checkGadgetOrModel(st *state.State, snapName string) error {
+	st.Lock()
+	defer st.Unlock()
+
+	var snapst snapstate.SnapState
+	if err := snapstate.Get(st, snapName, &snapst); err != nil {
+		return fmt.Errorf("failed to get snapstate for snap %s: %v", snapName, err)
+	}
+
+	// get the brand of the current snap
+	snapInfo, err := snapst.CurrentInfo()
```

Comment:
this will not have a the publisher set, it's only set when we get it as an answer from the store,  you need to do something like publisherAccount in daemon/snaps.go. We probably need to move that helper somewhere like in assertstate for example

Reply:
Ahh right, thanks for pointing this out - I've now moved the function to assertstate in commit https://github.com/snapcore/snapd/pull/11780/commits/1312690d03e71fef75e1eccbd9af939ffb7afeec